### ISSUE TITLE
docker-compose import not copying environment variables values but only names

### ISSUE THREAD

**Issue by surajssd** [openshift/origin#8860]
I tried the new feature of `oc import docker-compose`, after exporting, the template has environment variable names but not their values
##### Version

``` bash
$ openshift version
openshift v1.3.0-alpha.0-463-g687315d
kubernetes v1.3.0-alpha.1-331-g0522e63
etcd 2.3.0
```
##### Steps To Reproduce

Using this docker-compose file1

``` yaml
[vagrant@localhost ~]$ cat docker-compose.yml 
mariadb:
  image: centos/mariadb
  ports:
    - "3306"
  environment:
    MYSQL_ROOT_PASSWORD: wordpress
    MYSQL_DATABASE: wordpress
    MYSQL_PASSWORD: wordpress
    MYSQL_USER: wordpress
  container_name: mariadb

wordpress:
  image: wordpress
  ports:
    - "8080:80"
  links:
    - mariadb:mysql
  restart: always
```

Exported as:

``` yaml
[vagrant@localhost ~]$ oc import docker-compose -f docker-compose.yml --as-template=mytemplate -o yaml --config admin.kubeconfig
apiVersion: v1
kind: Template
metadata:
  annotations:
    app.generate.openshift.io/warnings: |-
      not all docker-compose fields were honored:
      * wordpress: restart is ignored - all pods are automatically restarted
  creationTimestamp: null
  name: mytemplate
objects:
- apiVersion: v1
  kind: ImageStream
  metadata:
    creationTimestamp: null
    name: mariadb
  spec:
    tags:
    - annotations:
        openshift.io/imported-from: centos/mariadb
      from:
        kind: DockerImage
        name: centos/mariadb
      generation: null
      importPolicy: {}
      name: latest
  status:
    dockerImageRepository: ""
- apiVersion: v1
  kind: DeploymentConfig
  metadata:
    creationTimestamp: null
    name: mariadb
  spec:
    replicas: 1
    selector:
      deploymentconfig: mariadb
    strategy:
      resources: {}
    template:
      metadata:
        creationTimestamp: null
        labels:
          deploymentconfig: mariadb
      spec:
        containers:
        - env:
          - name: MYSQL_DATABASE
          - name: MYSQL_PASSWORD
          - name: MYSQL_ROOT_PASSWORD
          - name: MYSQL_USER
          image: centos/mariadb
          name: mariadb
          ports:
          - containerPort: 3306
          resources: {}
    test: false
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - mariadb
        from:
          kind: ImageStreamTag
          name: mariadb:latest
      type: ImageChange
  status: {}
- apiVersion: v1
  kind: ImageStream
  metadata:
    creationTimestamp: null
    name: wordpress
  spec:
    tags:
    - annotations:
        openshift.io/imported-from: wordpress
      from:
        kind: DockerImage
        name: wordpress
      generation: null
      importPolicy: {}
      name: latest
  status:
    dockerImageRepository: ""
- apiVersion: v1
  kind: DeploymentConfig
  metadata:
    creationTimestamp: null
    name: wordpress
  spec:
    replicas: 1
    selector:
      deploymentconfig: wordpress
    strategy:
      resources: {}
    template:
      metadata:
        creationTimestamp: null
        labels:
          deploymentconfig: wordpress
      spec:
        containers:
        - image: wordpress
          name: wordpress
          ports:
          - containerPort: 80
          resources: {}
    test: false
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - wordpress
        from:
          kind: ImageStreamTag
          name: wordpress:latest
      type: ImageChange
  status: {}
- apiVersion: v1
  kind: Service
  metadata:
    creationTimestamp: null
    name: mysql
  spec:
    ports:
    - name: 3306-tcp
      port: 3306
      targetPort: 3306
    selector:
      deploymentconfig: mariadb
  status:
    loadBalancer: {}
- apiVersion: v1
  kind: Service
  metadata:
    creationTimestamp: null
    name: wordpress
  spec:
    ports:
    - name: 80-tcp
      port: 80
      targetPort: 80
    selector:
      deploymentconfig: wordpress
  status:
    loadBalancer: {}
```

Above you can see that in the first import the values in `DeploymentConfig` `mariadb` the container's env has:

``` yaml
- name: MYSQL_DATABASE
- name: MYSQL_PASSWORD
- name: MYSQL_ROOT_PASSWORD
- name: MYSQL_USER
```

env's without values.

**Comment by surajssd** [comment]
Also tried using second example

Using this docker-compose file
```yaml
[vagrant@localhost ~]$ cat docker-compose.yml 
postgresql:
  image: swordphilic/postgresql
  ports:
    - "5432"
  environment:
    DB_NAME: gitlab
    DB_PASS: gitlab
    DB_USER: gitlab

gitlab:
  image: swordphilic/gitlab
  ports:
    - "30000:80"
    - "30001:443"
    - "30002:22"
  links:
    - redis
    - postgresql
  environment:
    DB_TYPE: postgres
    DB_NAME: gitlab
    DB_PASS: gitlab
    DB_USER: gitlab

redis:
  image: swordphilic/redis
  ports:
    - "6379"
```

Exported as:
```yaml
[vagrant@localhost ~]$ oc import docker-compose -f docker-compose.yml --as-template=mytemplate -o yaml --config admin.kubeconfig
apiVersion: v1
kind: Template
metadata:
  creationTimestamp: null
  name: mytemplate
objects:
- apiVersion: v1
  kind: ImageStream
  metadata:
    creationTimestamp: null
    name: gitlab
  spec:
    tags:
    - annotations:
        openshift.io/imported-from: swordphilic/gitlab
      from:
        kind: DockerImage
        name: swordphilic/gitlab
      generation: null
      importPolicy: {}
      name: latest
  status:
    dockerImageRepository: ""
- apiVersion: v1
  kind: DeploymentConfig
  metadata:
    creationTimestamp: null
    name: gitlab
  spec:
    replicas: 1
    selector:
      deploymentconfig: gitlab
    strategy:
      resources: {}
    template:
      metadata:
        creationTimestamp: null
        labels:
          deploymentconfig: gitlab
      spec:
        containers:
        - env:
          - name: DB_NAME
          - name: DB_PASS
          - name: DB_TYPE
          - name: DB_USER
          image: swordphilic/gitlab
          name: gitlab
          ports:
          - containerPort: 80
          - containerPort: 443
          - containerPort: 22
          resources: {}
    test: false
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - gitlab
        from:
          kind: ImageStreamTag
          name: gitlab:latest
      type: ImageChange
  status: {}
- apiVersion: v1
  kind: ImageStream
  metadata:
    creationTimestamp: null
    name: postgresql
  spec:
    tags:
    - annotations:
        openshift.io/imported-from: swordphilic/postgresql
      from:
        kind: DockerImage
        name: swordphilic/postgresql
      generation: null
      importPolicy: {}
      name: latest
  status:
    dockerImageRepository: ""
- apiVersion: v1
  kind: DeploymentConfig
  metadata:
    creationTimestamp: null
    name: postgresql
  spec:
    replicas: 1
    selector:
      deploymentconfig: postgresql
    strategy:
      resources: {}
    template:
      metadata:
        creationTimestamp: null
        labels:
          deploymentconfig: postgresql
      spec:
        containers:
        - env:
          - name: DB_NAME
          - name: DB_PASS
          - name: DB_USER
          image: swordphilic/postgresql
          name: postgresql
          ports:
          - containerPort: 5432
          resources: {}
    test: false
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - postgresql
        from:
          kind: ImageStreamTag
          name: postgresql:latest
      type: ImageChange
  status: {}
- apiVersion: v1
  kind: ImageStream
  metadata:
    creationTimestamp: null
    name: redis
  spec:
    tags:
    - annotations:
        openshift.io/imported-from: swordphilic/redis
      from:
        kind: DockerImage
        name: swordphilic/redis
      generation: null
      importPolicy: {}
      name: latest
  status:
    dockerImageRepository: ""
- apiVersion: v1
  kind: DeploymentConfig
  metadata:
    creationTimestamp: null
    name: redis
  spec:
    replicas: 1
    selector:
      deploymentconfig: redis
    strategy:
      resources: {}
    template:
      metadata:
        creationTimestamp: null
        labels:
          deploymentconfig: redis
      spec:
        containers:
        - image: swordphilic/redis
          name: redis
          ports:
          - containerPort: 6379
          resources: {}
    test: false
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - redis
        from:
          kind: ImageStreamTag
          name: redis:latest
      type: ImageChange
  status: {}
- apiVersion: v1
  kind: Service
  metadata:
    creationTimestamp: null
    name: gitlab
  spec:
    ports:
    - name: 22-tcp
      port: 22
      targetPort: 22
    - name: 80-tcp
      port: 80
      targetPort: 80
    - name: 443-tcp
      port: 443
      targetPort: 443
    selector:
      deploymentconfig: gitlab
  status:
    loadBalancer: {}
- apiVersion: v1
  kind: Service
  metadata:
    creationTimestamp: null
    name: postgresql
  spec:
    ports:
    - name: 5432-tcp
      port: 5432
      targetPort: 5432
    selector:
      deploymentconfig: postgresql
  status:
    loadBalancer: {}
- apiVersion: v1
  kind: Service
  metadata:
    creationTimestamp: null
    name: redis
  spec:
    ports:
    - name: 6379-tcp
      port: 6379
      targetPort: 6379
    selector:
      deploymentconfig: redis
  status:
    loadBalancer: {}
```

**Comment by surajssd** [comment]
Now I realised the ports information in above example is also not copied, the compose file has 
```yaml
gitlab:
  image: swordphilic/gitlab
  ports:
    - "30000:80"
    - "30001:443"
    - "30002:22"
```

but in the imported template I don't see that value being mapped anywhere.

**Comment by smarterclayton** [comment]
Fixed by #8871

On May 13, 2016, at 7:30 AM, Dan McPherson <notifications@github.com> wrote:

Assigned #8860 <https://github.com/openshift/origin/issues/8860> to
@smarterclayton <https://github.com/smarterclayton>.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/issues/8860#event-659794646>
